x-pack/plugins/snapshot_restore/public/app/services/api.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { useEffect, useState } from 'react';
+import { API_BASE_PATH } from '../../../common/constants';
+import { AppStateInterface, useAppState } from './app_context';
+
+export const useFetch = ({
+  path,
+  method,
+  body,
+  interval,
+}: {
+  path: string;
+  method: string;
+  body?: any;
+  interval?: number;
+}) => {
+  const [
+    {
+      core: { http, chrome },
+    },
+  ] = useAppState() as [AppStateInterface];
+
+  const [error, setError] = useState<null | any>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [data, setData] = useState<any>([]);
+
+  const fetch = async () => {
```

Comment:
The consumer provides `method` to this hook so it can also be used to send `POST` etc requests, and "fetch" seems more geared towards retrieving data (it's also already part of the [global namespace](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)). How about renaming this to `request` and `useRequest` instead?